### ISSUE TITLE
Bulk refresh of all sites update information

### ISSUE THREAD

**Issue by flow-motion** [akeeba/panopticon#311]
May be my installation has failures, or may be it is a feature... 
When i open panopticon i have to click every sites refresh button one to see, if there is a joomla or an extension update. 
Wouldn´t be a button useful that asks all websites in panopticon for the latest infos and ift there are any updates to do?

Have a great time - and THANKS for your work!

**Comment by nikosdion** [comment]
Go to Panopticon, Administration, Tasks.

In the "– Site –" drop-down select System Task.

Among other things, you must see two tasks, `refreshinstalledextensions` and `refreshsiteinfo` with their status showing as a green checkmark icon and their last execution date shown to be in the last hour. Example:

<img width="1128" alt="Screenshot 2023-11-17 at 11 23 49" src="https://github.com/akeeba/panopticon/assets/256041/455537d8-96ce-477a-8829-a9fcb1141d44">

These tasks update the information Panopticon knows about a site's extension updates and Joomla! and PHP update information respectively.

If these tasks do not exist or are disabled, let me know; that would be why you have a problem.

Moreover, you **must** have already set up the Panopticon CRON job running once every minute. I

f you have already done that and you still see nothing getting updated check if there are other tasks currently running, for example tasks to update sites and extensions, or tasks to take backups and run the PHP file change. These tasks take priority. You can see that if the last execution time of the tasks I mentioned is over 1 hour into the past.

If this happens, create more CRON jobs. Each CRON job is identical to the CRON job you already have; just copy it over several times.

Remember, the more work you ask Panopticon to do at the same time, the more CRON jobs you need to have set up so that the work gets done in parallel rather than sequentially. It's like trying to empty a big tank of water. The more pipes (CRON jobs) you have, the faster the tank will empty of water (work gets done). 

I can't tell you what is the ideal number of CRON jobs, as it greatly depends on what you ask Panopticon to do. Long-running tasks like updating Joomla and extensions on many sites, running backups, and running the PHP File Change Scanner, increase the amount of work that needs to be done. Start adding CRON jobs until work gets done in a reasonable amount of time. Most of you with 10-20 sites will needs 2–3 CRON jobs. If you have particularly big sites and you run daily (or even more frequent) backups and the PHP File Change Scanner you might need one CRON job for every 2–3 sites you have set up.

Remember. More water in the tank (stuff you ask Panopticon to do), more pipes (CRON jobs) are needed to keep it from overflowing.

**Comment by flow-motion** [comment]
Hi Nikolas

I have these infos beneath the 2 tasks: 
refreshsiteinfo
Refresh Site Information
 Last execution on2023-11-17 10:45:01 CET
 Next execution on2023-11-17 11:00:00 CET
 Last execution run time00:00:01


AND

refreshinstalledextensions
Refresh Installed Extensions
 Last execution on2023-11-17 10:45:02 CET
 Next execution on2023-11-17 11:00:00 CET
 Last execution run time00:00:00
 
 Both enabled and green


I did the setup of the cron job with the URL thing. So - sorry, but - no idea how to add some

Do i have to have panopticon in browser, that the atualisation is running? or will it be refreshed also, if i am not into it.

**Comment by nikosdion** [comment]
Great. This means that you get all the update information from all of your sites. Therefore the main page shows you which sites have updates available.

Now, click on Overview, Extension Updates in the top menu — assuming you have already updated to 1.0.4. If you would please read the release notes of version 1.0.4 which I published yesterday you will see it is exactly what you are looking for.

**Comment by flow-motion** [comment]
That´s awesome... WOW 
I will test and if any things go wrong, i will let you know THANKS MAN

**Comment by nunoleite** [comment]
Hi!
Sorry to reopen this, but i'm having the same problem.
The list of extension are not updated, and if i click on update it will update and show the number of extensions that need to be updated.
![Image 001](https://github.com/akeeba/panopticon/assets/173281/05b00027-aa4e-4295-8597-81c14d5cf905)
![Image 002](https://github.com/akeeba/panopticon/assets/173281/00e73152-258b-4712-b2a6-12a7f78cf484)
![Image 003](https://github.com/akeeba/panopticon/assets/173281/a281bde1-b802-4b2f-b9ac-4f4488e31e41)

**Comment by nikosdion** [comment]
Everything works fine. What you describe is Joomla caching updates for up to 24 hours **AS IT SHOULD**. Clicking the force reload button bypasses the cache. 

For very detailed information, please do read https://github.com/akeeba/panopticon/wiki/Periodic-information-update

**Comment by nunoleite** [comment]
This is like this for weeks, but i opened that item details and it shows this...
![Image 007](https://github.com/akeeba/panopticon/assets/173281/e135b49d-2f05-4f7c-a2db-0a7f3b81f1ee)
![Image 008](https://github.com/akeeba/panopticon/assets/173281/a599371a-d912-4a45-91e4-87b775924a76)
So the problem maybe just with the listing

**Comment by nikosdion** [comment]
Please do hover your cursor over that icon. As you can see it tells you that 47 extensions _do not have an update site_. Therefore, they do not have updates. Therefore, the sites overview listing correctly tells you there are no extension updates pending.

The sites overview is just that: an _overview_. It only gives you the most important information. In the case of extensions it's the number of pending updates (or an indication if there are no pending updates), whether auto-updates are in the queue / executing, updates with missing download keys (because you can see the update but not install it), and whether the extensions update task is stuck (no updates will be installed in this case). It will not show you items with missing update sites, potentially buggy updates etc. The latter are tertiary _potential_ problems, not something you can always do something about.

For example, you may have some custom extensions which do not have an update site. No point bugging you about them. Some 3PDs package their extensions incorrectly (e.g. Tassos), meaning that a package with 25 extensions will only have _one_ extension with an update site, the other ones appearing as not having any update servers. Again, not something worth troubling you about.

**Comment by nunoleite** [comment]
Ok, i understood that.
But it should check if there are new updates from time to time, right?
It's not doing that.

![Image 001](https://github.com/akeeba/panopticon/assets/173281/f6ac6167-c46f-4d6d-b2ee-f037d102cd0c)

After pressing the refresh button... now i have 17 new updates

![Image 002](https://github.com/akeeba/panopticon/assets/173281/d331c784-75f6-40ec-8768-7129d5725006)
![Image 003](https://github.com/akeeba/panopticon/assets/173281/fd117de1-c557-4e79-b187-8860faab1d46)

**Comment by nikosdion** [comment]
Please do remember that Joomla! itself caches the update information for each extension for a length of time which might be anywhere from one hour to 24 hours. Depending on when was the last time Joomla! checked for updates and how long the caching time for the update information is set on that site, it is possible that one site "sees" an update to an extension, but another site does not "see" the update just yet.

Do keep in mind that when you click the user interface button to reload the extension update information you are telling Panopticon to ask Joomla! to override the updates cache, forcing it to recheck whether an update is available for each and every extension Joomla! has an update site for. This is very slow and hits the update servers of third party extensions hard. While it will show you updates faster you should not make a habit of using that button regularly. It not only slows down your site, it also causes operational problems on third party extension developers' sites who have to cope with a higher load of update checks. It's best for you to just wait for Joomla! to detect the updates when its updates cache expires, and Panopticon to eventually show them to you.

**Comment by nunoleite** [comment]
Now i understand that this is no automatic.

As i told before i have some websites in Panopticon list that were added weeks ago. And these are telling me that there are no updates, and i know that there are.
If passed some time and the Panopticon list doesn't force Joomla to check from let's say 1 or 2 days or maybe more, 7 days and i have to wait that Joomla makes it automatically, then this App is not really useful.
It just lists the websites, because i have to click manually these refresh buttons one by one to see what needs to me updated.

The goal should be to alert me when there are updates to Joomla and extensions.

**Comment by nikosdion** [comment]
It **IS** automatic on our end. It's that Joomla! bloody forgets looking for updates unless you visit its backend (so, we can say that it needs… a kick in the butt?). I will be releasing version 1.0.2 of the connectors which force Joomla! to recheck for updates when the cache timeout is reached.

**Comment by nunoleite** [comment]
This could be an option in Panopticon so we could define that time with a minimum of 24 hours for example.
This way the user could manage their needs.
Again, Thanks.

**Comment by nikosdion** [comment]
@nunoleite This is an option in Joomla! itself since Joomla! 1.6.0 which was released in 2010 – a little over thirteen years ago. Go to the extensions updates page and click on Options. 

**I DO NOT RECOMMEND CHANGING THIS SETTING FROM THE DEFAULT (6 HOURS)**. 6 hours strikes a good balance between seeing updates relatively soon enough, and not overwhelming third party extension developers' servers. Remember, bandwidth ain't free. If you hit the update servers hard we have to pay for it, _even when the extension is free of charge_.

**Comment by nunoleite** [comment]
I have installed connector 1.0.2 on 1 website and i'm getting error 500.
The other websites are no installing this connector automatically as they are defined to do so. "Install any version".
